mesonbuild/scripts/meson_install.py

```diff
@@ -168,6 +169,48 @@ def install_man(d):
         else:
             do_copyfile(full_source_filename, outfilename)
 
+def install_fs_links(d):
+    for source, to, type_ in d.fs_links:
+        to += os.path.splitext(source)[1]
+        print('Linking {} to {} as {} link"'.format(source, to, type_))
+
+        to = os.path.join(d.fullprefix, to)
+        append_to_log(to)
+        outdir = os.path.dirname(to)
+        if not os.path.isabs(source):
+            source = get_destdir_path(d, source)
+        os.makedirs(outdir, exist_ok=True)
+
+        # Neither symlink nor link have a force option
+        try:
+            os.unlink(to)
+        except FileNotFoundError:
+            pass
+
+        if type_ == 'symbolic':
+            # Symbolic links need to be created relative to the install
+            # directory not the build directory to ensure that the install is
+            # portable.
+            with chdir(outdir):
+                try:
+                    os.symlink(os.path.relpath(source, outdir),
+                               os.path.relpath(to, outdir))
+                except NotImplementedError:
+                    print('WARNING: Not creating link, symlinks not supported '
+                          'on Windows < 6.0 (Vista)')
+                except OSError as e:
+                    # On windows either administrative privilege is required, or
+                    # for windows 10 with Developer mode enabled (which is new
+                    # in the "Creators Update")
+                    if e.errno == errno.EPERM and sys.platform == 'win32':
+                        print('WARNING: Not creating link, creating symlinks '
+                              'on Windows require admin privilege or developer'
+                              ' mode to be enabled')
```

Comment:
Maybe we should add that as `type: symbolic-or-copy` or something? Or maybe another kwarg lets you specify what to do if symlinking fails; whether to ignore it, error out, or copy.